lib/inc/internal/util/version_parsing.hpp

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file
+ * Declares utility functions parsing version strings.
+ */
+#pragma once
+
+#include <string>
+#include <tuple>
+
+namespace facter { namespace util { namespace version_parsing {
+
+    /**
+     * Parses the major and minor kernel versions for all platforms except Linux.
+     * @param version The version to parse.
+     * @return Returns a tuple of major and minor versions.
+     */
+    std::tuple<std::string, std::string> parse_kernel_version(std::string const& version);
```

Comment:
Since it's internal, I'm fine with this for now. If we end up with more varieties we'll rename it.